### DISCUSSION TITLE
Database components

### DISCUSSION THREAD

**Discussion by Odonno** [microsoft/aspire#2961] [Q&A] [answered]
Hi,

I started working on a new Aspire component for a new generation database but then I re-read the [guidelines](https://github.com/dotnet/aspire/blob/main/src/Components/README.md) and one thing clicked:

```
In most cases we expect that it is widely used, although we expect that part will be a judgement call.
```

I suppose that most databases in the database world are not well-known/widely used apart from the big ones (MySQL, SQL Server, Postgres, Oracle, MongoDb, Cassandra, Redis and Neo4j).

So my question is, should we have a limit in to what database Aspire should support? Considering things like Firestore (localhost emulator at most), ArangoDb, CockroachDB, etc...

**Comment by davidfowl** [comment, marked as answer]
We expect to support all databases, but they won’t be part of the initial release of part of this repository. Once we GA, many of these components will live outside of this project and others will be able to build components and resources and publish them to nuget.

What we did was attempt to bootstrap the project and ecosystem to figure out what patterns would be effective for a wide variety of libraries. Now that we’ve done that, the work will be figuring out how to make it more distributed.

**Comment by robertstefan** [comment]
Any updates regarding this @Odonno ?

**Reply by davidfowl** [reply]
https://github.com/CommunityToolkit/Aspire is a great place for integrations (renamed from components)
